clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -75,6 +74,7 @@
     private final int autoCommitIntervalMs;
     private final ConsumerInterceptors<?, ?> interceptors;
     private final boolean excludeInternalTopics;
+    private final AtomicInteger pendingAsyncCommits;
```

Comment:
I'm not actually sure we need this. The method `ConsumerNetworkClient.pendingRequestCount` handles sent and unset requests, so it seems sufficient to just wait until the count reaches zero to be sure all outstanding offset commits have been sent.